spanner/client.go

```diff
@@ -549,6 +571,43 @@ func allClientOpts(numChannels int, compression string, userOpts ...option.Clien
 	return append(allDefaultOpts, userOpts...)
 }
 
+func unaryPeerSetter() grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		p, ok := peer.FromContext(ctx)
+		if ok {
+			opts = append(opts, grpc.Peer(p))
+		}
+		return invoker(ctx, method, req, reply, cc, opts...)
+	}
+}
+
+// streamPeerSetter makes the grpc connection include peer information in a context variable keyed by PeerKey{} if it exists.
+func streamPeerSetter() grpc.StreamClientInterceptor {
+	return func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string,
+		streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+		p, ok := peer.FromContext(ctx)
+		if ok {
+			opts = append(opts, grpc.Peer(p))
+		}
+		return streamer(ctx, desc, cc, method, opts...)
+	}
+}
+
+// AddUnaryPeerInterceptor intercepts unary requests and add PeerKey.
+func AddUnaryPeerInterceptor() grpc.UnaryClientInterceptor {
+	unaryInterceptors := []grpc.UnaryClientInterceptor{}
+	unaryInterceptors = append(unaryInterceptors, unaryPeerSetter())
+	return go_grpc_middleware.ChainUnaryClient(unaryInterceptors...)
```

Comment:
grpc-go already have `WithChainUnaryInterceptor` and `WithChainStreamInterceptor`. No need `go_grpc_middleware` dependency.

- https://pkg.go.dev/google.golang.org/grpc@v1.65.0 #WithChainUnaryInterceptor
- https://pkg.go.dev/google.golang.org/grpc@v1.65.0#WithChainStreamInterceptor